### PR TITLE
AWS and GCE deployment

### PR DESCRIPTION
**Pull request by edevenport**
This adds GCE deployment support, status updates, and general clean up.

### CODE REVIEW

File: ansible/roles/build-cloud-infrastructure/tasks/provider/gce.yml [hunk 1]
```diff
@@ -1 +1,18 @@
...
+    dest: ./gce_credentials.json
+
+- name: Create firewall rules
+  gce_net:
+    name: default
+    fwname: netsil-fw
+    allowed: 'tcp:22,80,443,2001,2003;udp:2003'
+    project_id: "{{ gce_project_id }}"
+    service_account_email: "{{ gce_service_account_email }}"
+    credentials_file: ./gce_credentials.json
```
Comment (luhkevin) [inline ansible/roles/build-cloud-infrastructure/tasks/provider/gce.yml]: Shouldn't this be `{{ gce_credentials_file }}`?

File: ansible/roles/build-cloud-infrastructure/tasks/provider/gce.yml [hunk 2]
```diff
@@ -1 +1,18 @@
...
+- name: Create GCE credentials file
+  copy:
+    content: "{{ gce_credentials_file }}"
+    dest: ./gce_credentials.json
+
+- name: Create firewall rules
+  gce_net:
+    name: default
+    fwname: netsil-fw
+    allowed: 'tcp:22,80,443,2001,2003;udp:2003'
```
Comment (luhkevin) [inline ansible/roles/build-cloud-infrastructure/tasks/provider/gce.yml]: Could you also add tcp 2000 to this list? We have a new architecture change where that port is used by default.

File: ansible/roles/build-cloud-infrastructure/tasks/provider/gce.yml (hunk 1 above)
Comment (luhkevin) [inline ansible/roles/build-cloud-infrastructure/tasks/provider/gce.yml]: Oh never mind, I see later in the code that it's copied to `./gce_credentials.json`

File: ansible/roles/build-cloud-infrastructure/tasks/provider/gce.yml (hunk 2 above)
Comment (luhkevin) [inline ansible/roles/build-cloud-infrastructure/tasks/provider/gce.yml]: (The same for the AWS rules, I suppose)

File: ansible/roles/build-cloud-infrastructure/tasks/provider/gce.yml (hunk 1 above)
Comment (edevenport) [inline ansible/roles/build-cloud-infrastructure/tasks/provider/gce.yml]: Correct. `{{ gce_credentials_file }}` is the content of the file and then we create the `./gce_credentials.json` file with the content. The filename could have been set as a variable as well, but I'm not sure it's necessary.

File: ansible/roles/build-cloud-infrastructure/tasks/provider/gce.yml (hunk 2 above)
Comment (edevenport) [inline ansible/roles/build-cloud-infrastructure/tasks/provider/gce.yml]: TCP port 2000 has been added to both GCE and AWS.

**Pull request merged**